stl/src/tzdb.cpp

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <internal_shared.h>
+#include <xtzdb.h>
+
+#include <Windows.h>
+
+#pragma comment(lib, "Advapi32")
+
+_EXTERN_C
+
+_RegistryLeapSecondInfo* __stdcall __std_tzdb_get_reg_leap_seconds(
+    const size_t prev_reg_ls_size, size_t* current_reg_ls_size) {
+    // On exit---
+    //    *current_reg_ls_size <= prev_reg_ls_size, *reg_ls_data == nullptr --> no new data
+    //    *current_reg_ls_size >  prev_reg_ls_size, *reg_ls_data != nullptr --> new data, successfully read
+    //    *current_reg_ls_size == 0,                *reg_ls_data != nullptr --> new data, failed reading
+    //    *current_reg_ls_size >  prev_reg_ls_size, *reg_ls_data == nullptr --> new data, failed allocation
+
+    constexpr auto reg_key_name    = TEXT("SYSTEM\\CurrentControlSet\\Control\\LeapSecondInformation");
+    constexpr auto reg_subkey_name = TEXT("LeapSeconds");
+    *current_reg_ls_size           = 0;
+    HKEY leap_sec_key              = 0;

```

Comment:
@mnatsuhara found that `HKEY` is a `HANDLE`, so we should initialize it with `nullptr` instead of `0`-as-null.